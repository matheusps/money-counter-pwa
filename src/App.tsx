import React, { FC, useState } from 'react'
import {
  DarksideTheme,
  Surface,
  Button,
  Heading,
  Emoji,
  Input,
} from 'darkside-ui'

import useStore from './store/useStore'
const URL = 'http://localhost:5000/predict'

const App: FC = () => {
  const [state, dispatch] = useStore()
  const [image, setImage] = useState('')

  function base64(file, callback) {
    const result = {
      base64: '',
    }
    const reader = new FileReader()
    reader.onload = e => {
      //@ts-ignore
      const base64 = btoa(e.target.result)
      result.base64 = base64
      callback(result)
    }
    reader.readAsBinaryString(file)
  }

  const handleFileChange = event => {
    const { target } = event
    const { files } = target
    base64(files[0], data => {
      console.log(data.base64)
      setImage(data.base64)
    })
  }

  const sendPicture = () => {
    fetch(URL, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
      },
      body: image,
    })
      .then(response => response.json())
      .then(responseJson => {
        console.log(responseJson)
      })
      .catch(e => console.log(e))
  }

  return (
    <DarksideTheme>
      <Surface
        height="90vh"
        direction="column"
        justify="space-between"
        items="center"
        style={{ paddingTop: '3rem' }}
      >
        <Emoji size="xl" symbol={state.emoji} />
        <Heading style={{ textAlign: 'center' }} level={6}>
          {state.message}
        </Heading>
        <Input type="file" onChange={handleFileChange} />
        <Button size="xl" onClick={() => sendPicture()} full>
          {state.btnText}
        </Button>
      </Surface>
    </DarksideTheme>
  )
}

export default App
