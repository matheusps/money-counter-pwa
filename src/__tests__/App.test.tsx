import React from 'react'
import { render } from 'react-testing-library'
import 'jest-styled-components'

import App from '../App'

describe('App component', () => {
  const component = render(<App />)

  it('should render', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
