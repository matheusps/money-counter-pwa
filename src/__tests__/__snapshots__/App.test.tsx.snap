// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  padding: 3rem;
  background-color: #FEFEFE;
}

.c1 {
  color: #3165EC;
}

.c2 {
  background-color: #F0F6FF;
  color: #3165EC;
  width: 100%;
  font-size: 1.5rem;
  padding: 0.8rem 2.6rem 0.8rem 2.6rem;
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  font-weight: 500;
  font-stretch: normal;
  line-height: 1.4;
  margin: 0.2rem;
  display: relative;
  overflow: hidden;
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: -webkit-transform 0.2s ease-in;
  -webkit-transition: transform 0.2s ease-in;
  transition: transform 0.2s ease-in;
  will-change: transform;
}

.c2:hover {
  -webkit-transform: scale(0.975);
  -ms-transform: scale(0.975);
  transform: scale(0.975);
  -webkit-transform: scale(0.975);
}

.c2:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
  -webkit-transform: scale(0.9);
}

.c2:focus {
  outline: none;
}

<div
    class="c0"
  >
    <img
      class=""
      src="camera.png"
    />
    <h2
      class="c1"
    >
      Take or send a picture
    </h2>
    <button
      class="c2"
    >
      Send
    </button>
  </div>
</DocumentFragment>
`;
